### PR TITLE
Exposing Headers (BrowserHeaders)

### PR DESCRIPTION
**Pull request by MarcusLongmuir**
This PR introduces the first dependency (https://github.com/improbable-eng/js-browser-headers) in order to expose the `Headers` instance that `fetch` returns (or is created by parsing the headers as a string).

`BrowserHeaders` provides a consistent API between browsers, which is in keeping with the intention of this package which is to abstract the different implementations.

This PR also changes to using Webpack to build the tests and adds source map support so that you can see stack traces for the test failures.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,6 +78,19 @@ A function which implements the following interface:
...
 The `onChunk` handler will be invoked each time a chunk of data it returned by the server. This function will be invoked one or more times depending on the response.  The function is invoked with two arguments; the first is an optional error which will be null unless there was a parsing error thrown by the `chunkParser``.  The second argument is an optional parsedChunk value which is produced by the supplied `chunkParser` (see: `options.chunkParser`).
 
+#### onHeaders (optional)
+A function which implements the following interface:
+
+```js
+(headers, statusCode) => undefined
+```
+
+A function which will be invoked once when the browser has returned the headers of the response. This function is invoked with two arguments:
```
Comment (jonny-improbable) [inline README.md]: please can you add a note on the temporal coupling here (ie: does this get invoked before the first chunk?)

File: src/impl/fetch.js [hunk 2]
```diff
@@ -35,7 +37,11 @@ export default function fetchRequest(options) {
   }
 
   fetch(options.url, { headers, method, body, credentials })
-    .then(res => pump(res.body.getReader(), res))
+    .then(res => {
+      const browserHeaders = new BrowserHeaders(res.headers);
```
Comment (jonny-improbable) [inline src/impl/fetch.js]: pls remove the temp.

File: src/impl/mozXhr.js [hunk 3]
```diff
@@ -15,6 +17,12 @@ export default function mozXhrRequest(options) {
     });
   }
 
+  function onStateChange() {
+    if(this.readyState == this.HEADERS_RECEIVED) {
+      const browserHeaders = new BrowserHeaders(this.getAllResponseHeaders());
```
Comment (jonny-improbable) [inline src/impl/mozXhr.js]: again please remove the temp.

File: src/impl/xhr.js [hunk 4]
```diff
@@ -21,6 +23,13 @@ export default function xhrRequest(options) {
     });
   }
 
+  function onStateChange() {
+    if(this.readyState == this.HEADERS_RECEIVED) {
+      const browserHeaders = new BrowserHeaders(this.getAllResponseHeaders());
```
Comment (jonny-improbable) [inline src/impl/xhr.js]: again please remove the temp.

**Comment by MarcusLongmuir** [comment]
Don't merge this yet. Still wrestling with an iterator issue in the BrowserHeader dependency.

**Review by jonnyreeves — APPROVED** [review]

**Pull request merged**